common/src/View/CompilationRunner.cpp

```diff
@@ -140,31 +141,29 @@ namespace TrenchBroom {
 
         void CompilationRunToolTaskRunner::doTerminate() {
             if (m_process != nullptr) {
-                disconnect(m_process.get(), &QProcess::errorOccurred, this, &CompilationRunToolTaskRunner::processErrorOccurred);
-                disconnect(m_process.get(), QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished), this, &CompilationRunToolTaskRunner::processFinished);
+                disconnect(m_process, &QProcess::errorOccurred, this, &CompilationRunToolTaskRunner::processErrorOccurred);
+                disconnect(m_process, QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished), this, &CompilationRunToolTaskRunner::processFinished);
                 m_process->kill();
-                m_process.reset();
                 m_context << "\n\n#### Terminated\n";
             }
         }
 
         void CompilationRunToolTaskRunner::startProcess() {
             assert(m_process == nullptr);
 
+            emit start();
             try {
-                const auto toolPath = IO::Path(interpolate(m_task->toolSpec()));
-                const auto parameters = interpolate(m_task->parameterSpec());
-                const auto cmd = toolPath.asString() + " " + parameters;
                 const auto workDir = m_context.variableValue(CompilationVariableNames::WORK_DIR_PATH);
+                const auto cmd = this->cmd();
 
                 m_context << "#### Executing '" << cmd << "'\n";
 
                 if (!m_context.test()) {
-                    m_process = std::make_unique<QProcess>(this);
-                    connect(m_process.get(), &QProcess::errorOccurred, this, &CompilationRunToolTaskRunner::processErrorOccurred);
-                    connect(m_process.get(), QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished), this, &CompilationRunToolTaskRunner::processFinished);
-                    connect(m_process.get(), &QProcess::readyReadStandardError, this, &CompilationRunToolTaskRunner::processReadyReadStandardError);
-                    connect(m_process.get(), &QProcess::readyReadStandardOutput, this, &CompilationRunToolTaskRunner::processReadyReadStandardOutput);
+                    m_process = new QProcess(this);
```

Comment:
Nice. I was going to comment about the original code that having 2 ownership systems in parallel `std::make_unique`  + QObject was making things more confusing than necessary, even if it does technically work.

Reply:
Yeah, that was just plain wrong, I don't think I understood Qt's ownership semantics when I wrote this.